experiments/language_rotation_sweep_fr_1.py

```diff
@@ -0,0 +1,224 @@
+                    "uncentered_linear_map",
+                    "biased_uncentered_linear_map",
+                    "rotation",
+                    "biased_rotation",
+                    "uncentered_rotation",
+                ]
+            },
+            "lr": {"distribution": "uniform", "min": 0, "max": 0.1},
+            "weight_decay": {"distribution": "uniform", "min": 0, "max": 0.01},
+        },
+    }
+
+    if no_processing:
+        model = tl.HookedTransformer.from_pretrained_no_processing(model_name)
+    else:
+        model = tl.HookedTransformer.from_pretrained(model_name)
+
+    datasets_folder = repo_path_to_abs_path("datasets")
+    file_path = f"{datasets_folder}/wikdict/2_extracted/eng-fra.json"
+    with open(file_path, "r") as file:
+        word_pairs = json.load(file)
+    random.seed(1)
+    random.shuffle(word_pairs)
+    split_index = int(len(word_pairs) * 0.97)
+    train_en_fr_pairs = word_pairs[:split_index]
+    test_en_fr_pairs = word_pairs[split_index:]
+
+    train_word_pairs = filter_word_pairs(
+        model,
+        train_en_fr_pairs,
+        discard_if_same=True,
+        min_length=4,
+        # capture_diff_case=True,
+        capture_space=True,
+        capture_no_space=True,
+        print_pairs=True,
+        print_number=True,
+        max_token_id=200_000,
+        # most_common_english=True,
+        # most_common_french=True,
+    )
+
+    test_word_pairs = filter_word_pairs(
+        model,
+        test_en_fr_pairs,
+        discard_if_same=True,
+        min_length=4,
+        # capture_diff_case=True,
+        capture_space=True,
+        capture_no_space=True,
+        # print_pairs=True,
+        print_number=True,
+        max_token_id=200_000,
+        # most_common_english=True,
+        # most_common_french=True,
+    )
+
+    train_en_toks, train_fr_toks, _, _ = tokenize_word_pairs(model, train_word_pairs)
+    test_en_toks, test_fr_toks, _, _ = tokenize_word_pairs(model, test_word_pairs)
+
+    if layernorm:
+        train_en_embeds = t.nn.functional.layer_norm(
+            model.embed.W_E[train_en_toks].detach().clone(), [model.cfg.d_model]
+        )
+        train_fr_embeds = t.nn.functional.layer_norm(
+            model.embed.W_E[train_fr_toks].detach().clone(), [model.cfg.d_model]
+        )
+        test_en_embeds = t.nn.functional.layer_norm(
+            model.embed.W_E[test_en_toks].detach().clone(), [model.cfg.d_model]
+        )
+        test_fr_embeds = t.nn.functional.layer_norm(
+            model.embed.W_E[test_fr_toks].detach().clone(), [model.cfg.d_model]
+        )
+    else:
+        train_en_embeds = model.embed.W_E[train_en_toks].detach().clone()
+        train_fr_embeds = model.embed.W_E[train_fr_toks].detach().clone()
+        test_en_embeds = model.embed.W_E[test_en_toks].detach().clone()
+        test_fr_embeds = model.embed.W_E[test_fr_toks].detach().clone()
+        # all of these are shape [batch, pos, d_model]
+
+    def train():
```

Comment:
I generally prefer to avoid defining functions inside of functions where possible